Including gap headers fails if a garbage collector is not specified

### Observed behaviour
Try to compile the following file `testgap.c' 
```
#include <gap/objects.h>
```
with `gcc testgap.c`:
```
$ gcc testgap.c
In file included from /usr/include/gap/objects.h:20,
                 from testgap.c:1:
/usr/include/gap/gasman.h:401:2: error: #error unknown garbage collector
  401 | #error unknown garbage collector
      |  ^~~~~
```

### Expected behaviour

File compiles without needing to specify a garbage collector

### Copy and paste GAP banner (to tell us about your setup)
```
┌───────┐   GAP 4.12.0 of 2022-08-18
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv8
 Configuration:  gmp 6.2.1, GASMAN, readline
```

--- a/src/common.h
+++ b/src/common.h
@@ -35,6 +35,10 @@
 // compiled with the wrong ABI
 GAP_STATIC_ASSERT(sizeof(void *) == SIZEOF_VOID_P, "sizeof(void *) is wrong");
 
+// if no GC is selected explicitly, default to GASMAN
+#if !defined(USE_BOEHM_GC) && !defined(USE_JULIA_GC) && !defined(USE_GASMAN)
+#define USE_GASMAN 1
+#endif
 
 // check for cygwin
 #if defined(__CYGWIN__) || defined(__CYGWIN32__)

